src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -208,13 +208,12 @@ private static bool IsContentTransform(JsonSchema schema)
         string? propertyPath)
     {
         schemaMap.TryGetSchema(obj, out var schema);
-        var href = GetXrefHref(file, uid, uidCount, obj.Parent == null);
+        var href = GetXrefHref(file, uid, uidCount, string.IsNullOrEmpty(propertyPath));
         var monikers = _monikerProvider.GetFileLevelMonikers(errors, file);
         var schemaType = GetSchemaType(uidSchema.SchemaType, schema?.SchemaTypeProperty, propertyPath, obj, file);
 
         var xref = new InternalXrefSpec(uid, href, file, monikers)
         {
-            DeclaringPropertyPath = obj.Parent?.Path,
```

Comment:
Does `DeclaringPropertyPath` always equal to `SchemaType`?

Reply:
No, but the `DeclaringPropertyPath` here is only used to determine if it is an achievement, which could be implemented using `SchemaType`.

Reply:
Or yes for `badge` and `trophy` which are the only things we care.